### DISCUSSION TITLE
Persistence

### DISCUSSION THREAD

**Discussion by jtothebell** [jtothebell/fake-08#62] [General]
New Features:
*Add settings ini file to store preferred screen stretch.
*Read and write Cart Data to files (adds save data capability)
*Add partial stat implementation

General Fixes
*add mapdraw alias to map function (fixes some old carts crashing)
*update z8lua split, pairs, ipairs, and tonum to handle args how pico 8 does (fixes several cart crashes)
*fix infinite loop when trying to draw an empty oval
*fix cursor location after calling print with a location
*use corretly mapped color in print calls
*fix handling of negative length passed to memset/memcpy/reload
*fix _update and _draw calls if they change at runtime
*leave general use ram alone after a reload call

Vita:
*Improve perf by utilizing overclock

<hr /><em>This discussion was created from the release <a href='https://github.com/jtothebell/fake-08/releases/tag/v0.0.2.6'>Persistence</a>.</em>

**Comment by hippydave** [comment]
Awesome to see 😎

**Comment by hippydave** [comment]
Oh let me know if I can help with Switch testing

**Reply by jtothebell** [reply]
Thanks! I can't test thoroughly now, but if you'd like to, I think this build should have the issues resolved (input was messed up by a refactor, and I guess I forgot to test it until after publishing)
[FAKE-08-switch.zip](https://github.com/jtothebell/fake-08/files/6305810/FAKE-08-switch.zip)

**Reply by hippydave** [reply]
Thanks for that, the input seems to be working fine. I noticed for the one cart I tried, the cart data was saved, but not reloaded.

**Reply by jtothebell** [reply]
Which cart was it? I found one bug today that could have prevented some data from saving.

**Reply by hippydave** [reply]
It was actually my own old game, Cellephant https://www.lexaloffle.com/bbs/?tid=29736

**Reply by jtothebell** [reply]
Oh, thanks. That turns out to be a separate issue. The Pico-8 wiki doesn't mention `cartdata` returning a boolean, but the manual does. I wasn't returning anything, so in Cellephant the check `if cartdata("bittwist_cellephant")` would always be falsy. Just pushed a fix if you want to try it (you can also download the artifact of the latest action):
[FAKE-08-switch.zip](https://github.com/jtothebell/fake-08/files/6318683/FAKE-08-switch.zip)

**Reply by hippydave** [reply]
That fixed it, good job!
